l10n_it_riba/migrations/16.0.1.0.0/pre-migrate.py

```diff
@@ -1,284 +1,24 @@
-        ),
-        (
-            "riba.configuration",
-            "past_due_journal_id",
-        ),
-    ],
-    [
-        (
-            "riba.accreditation",
-            "accreditation_journal_id",
-        ),
-        (
-            "riba.credit",
-            "credit_journal_id",
-        ),
-    ],
-    [
-        (
-            "riba.accreditation",
-            "accreditation_account_id",
-        ),
-        (
-            "riba.credit",
-            "credit_account_id",
-        ),
-    ],
-    [
-        (
-            "riba.accreditation",
-            "accreditation_amount",
-        ),
-        (
-            "riba.credit",
-            "credit_amount",
-        ),
-    ],
-    [
-        (
-            "riba.unsolved",
-            "unsolved_journal_id",
-        ),
-        (
-            "riba.past_due",
-            "past_due_journal_id",
-        ),
-    ],
-]
-RENAMED_XMLIDS = [
-    ("seq_riba_distinta", "seq_riba_slip"),
-    ("print_distinta_qweb", "print_slip_qweb"),
-    ("access_riba_distinta_uinvoice", "access_riba_slip_uinvoice"),
-    ("access_riba_distinta_group_invoice", "access_riba_slip_group_invoice"),
-    ("access_riba_distinta_user", "access_riba_slip_user"),
-    ("access_riba_distinta_accountant", "access_riba_slip_accountant"),
-    ("access_riba_distinta_line_uinvoice", "access_riba_slip_line_uinvoice"),
-    ("access_riba_distinta_line_group_invoice", "access_riba_slip_line_group_invoice"),
-    ("access_riba_distinta_line_user", "access_riba_slip_line_user"),
-    ("access_riba_distinta_line_accountant", "access_riba_slip_line_accountant"),
-    ("access_riba_distinta_move_line_uinvoice", "access_riba_slip_move_line_uinvoice"),
-    (
-        "access_riba_distinta_move_line_group_invoice",
-        "access_riba_slip_move_line_group_invoice",
-    ),
-    ("access_riba_distinta_move_line_user", "access_riba_slip_move_line_user"),
-    (
-        "access_riba_distinta_move_line_accountant",
-        "access_riba_slip_move_line_accountant",
-    ),
-    ("access_riba_unsolved", "access_riba_past_due"),
-    ("access_riba_accreditation", "access_riba_credit"),
-    ("riba_distinta_company_rule", "riba_slip_company_rule"),
-    ("riba_distinta_line_company_rule", "riba_slip_line_company_rule"),
-    ("view_riba_da_emettere_tree", "view_riba_to_issue_tree"),
-    ("action_riba_da_emettere", "action_riba_to_issue"),
-    ("menu_riba_da_emettere", "menu_riba_to_issue"),
-    ("view_distinta_riba_filtri", "view_slip_riba_filter"),
-    ("view_distinta_riba_tree", "view_slip_riba_tree"),
-    ("view_riba_distinta_line_form", "view_riba_slip_line_form"),
-    ("distinta_riba_action", "slip_riba_action"),
-    ("distinta_layout", "slip_layout"),
-    ("distinta_qweb", "slip_qweb"),
-    ("riba_accreditation", "riba_credit"),
-    ("riba_accreditation_action", "riba_credit_action"),
-    ("riba_unsolved", "riba_past_due"),
-    ("riba_unsolved_action", "riba_past_due_action"),
-]
+from . import hooks
```

Comment:
Ho provato la migrazione con OpenUpgrade, all'esecuzione di questa riga ottengo l'errore:
> ImportError: attempted relative import with no known parent package

<details><summary>Stack trace</summary>
<pre>
<code>
2024-06-06 08:26:33,369 60927 ERROR odoo16-mig-l10n_it_riba odoo.modules.migration: module l10n_it_riba: Unable to load pre-migration file /path/to/odoo16/l10n-italy/l10n_it_riba/migrations/16.0.1.0.0/pre-migrate.py 
Traceback (most recent call last):
  File "/path/to/odoo16/odoo/odoo/modules/migration.py", line 180, in migrate_module
    mod = load_script(pyfile, name)
  File "/path/to/odoo16/odoo/odoo/modules/migration.py", line 25, in load_script
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/path/to/odoo16/l10n-italy/l10n_it_riba/migrations/16.0.1.0.0/pre-migrate.py", line 7, in <module>
    from . import hooks
ImportError: attempted relative import with no known parent package
</code>
</pre>
</details>

Reply:
A me non era capitato questo errore, però da v15 a v16 ho usato la migrazione del core di Odoo SA e poi ho lanciato un `-u all` con tutti i moduli disponibili.
Mi sembra che anche altri moduli facciano degli import simili, secondo te devo mettere tutto il percorso `from odoo.addons.l10n_it_riba`?

Reply:
> Mi sembra che anche altri moduli facciano degli import simili

Ad esempio? Probabilmente li fanno in punti dove c'è un `parent package`.

> secondo te devo mettere tutto il percorso `from odoo.addons.l10n_it_riba`?

Sì, l'esempio in https://github.com/OCA/l10n-italy/wiki/Migrazione-con-rinomina-modulo,-compatibile-con-OpenUpgrade-e-Odoo-SA infatti fa così

Reply:
Se modifico l'import come dicevamo ottengo questo warning di pylint: 
![image](https://github.com/OCA/l10n-italy/assets/93990941/bd41ebff-087d-43bd-9cc0-8e8e0805ab53)

Però effettivamente anche in altre parti è usato l'import assoluto come dicevi

Reply:
Qua effettivamente è stato usato l'import assoluto, non so se nella 16 pylint è più restrittivo: https://github.com/OCA/l10n-italy/blob/1194fb209a8f97410cbe780251de72637dde7d0d/l10n_it_financial_statements_report/migrations/14.0.1.0.0/pre-migrate.py

Reply:
Vedo che sei riuscito in quella PR provvisoria, ma a me da questo errore:

![image](https://github.com/OCA/l10n-italy/assets/93990941/1ae33bdd-13e8-4938-abb9-981d8be8da2a)

Reply:
> nella 16 pylint è più restrittivo

Sembra di sì.
Quindi o dici a pylint di non controllare quella riga (come fatto in https://github.com/OCA/l10n-italy/blob/4df2c2b109763c287a307679a4ddbcdda7442d56/l10n_it_asset_management/migrations/16.0.1.0.0/pre-migrate.py#L4), o trovi un modo che vada a bene a Pylint.
Di sicuro non si può lasciare com'è adesso perché dà errore, che ne pensi?

Reply:
Hai ragione, alla fine ho fatto come hai indicato in `l10n_it_asset_management`

Reply:
:+1: